### PR TITLE
Remove Servlet dependencies from VievEngineContext

### PR DESCRIPTION
**Pull request by ivargrimstad**
HttpServletRequest and HttpServletResponse removed from ViewEngineContext.
Implementation refactored according to https://github.com/mvc-spec/mvc-spec/pull/3

This fix will #close [OZARK-76](https://java.net/jira/browse/OZARK-76)

### CODE REVIEW

File: ozark/src/main/java/org/glassfish/ozark/engine/ServletViewEngine.java [hunk 1]
```diff
@@ -101,7 +102,7 @@ protected void forwardRequest(ViewEngineContext context, String... extensions)
                 rd = servletContext.getNamedDispatcher(e.getKey());     // by servlet name
 
                 // Need new request with updated URI and extension matching semantics
-                request = new HttpServletRequestWrapper(context.getRequest()) {
+                request = new HttpServletRequestWrapper(((ViewEngineContextImpl) context).getRequest()) {
```
Comment (Turini) [inline ozark/src/main/java/org/glassfish/ozark/engine/ServletViewEngine.java]: I'd just extract a method to avoid repeat the casting, what do you think?

**Comment by Turini** [comment]
that sounds great, thanks @ivargrimstad. +1 for merging it

File: ozark/src/main/java/org/glassfish/ozark/engine/ServletViewEngine.java (hunk 1 above)
Comment (ivargrimstad) [inline ozark/src/main/java/org/glassfish/ozark/engine/ServletViewEngine.java]: Yes, just wanted to show it works first. Can clean up later.

**Comment by chkal** [comment]
Hey @ivargrimstad. What do you think about closing this one. It doesn't merge any more, the tests are red. Not sure if it makes sense to keep it open. We can open a new one later to address this.

**Comment by ivargrimstad** [comment]
Yes, It's been hanging around way to far. Pretty easy to fix later anyway. I'll close it.

**Pull request closed without merging**